app/graphql/types/query_type.rb

```diff
@@ -1,23 +1,25 @@
 module Types
   class QueryType < Types::BaseObject
+    # Return all markets
     field :all_markets, [Types::MarketType], null: false
     def all_markets
       Market.all
     end
-
-    # Save for future use to search markets by products, etc.
-    # 
-    # field :all_markets, [Types::MarketType], null: false
-    # def all_markets
-    #   Market.all
-    # end
-
+    # Return a single market by id
     field :market, Types::MarketType, null: false do
       argument :id, Integer, required: true
     end
-    def market(id)
-      Market.find(id[:id])
+    def market(id:)
+      Market.find(id)
+    end
+    # Return markets by location
+    field :markets_by_location, [Types::MarketType], null: false do
+      argument :lat, Float, required: true
+      argument :lng, Float, required: true
+      argument :radius, Integer, required: true
+    end
+    def markets_by_location(lat:, lng:, radius:)
```

Comment:
same here that's all that was needed to get this method running

Reply:
One thing I was thinking of is probably that will eventually want a resolver to handle bad requests, but we can update that soon